src/classes/AiActions/AiHandler.js

```diff
@@ -0,0 +1,98 @@
+ * that the Ai will execute.
+ */
+export default class Handler {
+  constructor (event) {
+    this.move = new AiMove()
+    this.opponentPO = this.move.getOpponentToAttack(event, 'POWEROUTAGE')
+    this.opponentVirus = this.move.getOpponentToAttack(event, 'VIRUS')
+    let hand = this.move.organizeHand(event)
+    let boolSide = store.getters.getCoinMsg
+    let canGroup = this.move.findGroup(event.stack, hand.bestGCard)
+    this.antiVirus = new AntiVirus(hand, boolSide, this.move, event)
+    this.firewall = new Firewall(hand, boolSide, this.move, event)
+    this.generator = new Generator(hand, boolSide, this.move, event)
+    this.overclock = new OverClock(hand, boolSide, this.move, event)
+    this.batteryBackup = new BatteryBackup(hand, boolSide, this.move, event)
+    this.group = new Group(hand, boolSide, this.move, event, canGroup)
+    this.virus = new Virus(hand, boolSide, this.move, event, this.opponentVirus)
+    this.hack = new Hack(hand, boolSide, this.move, event)
+    this.powerOutage = new PowerOutage(hand, boolSide, this.move, event, this.opponentPO)
+    this.instruction = new Instruction(hand, boolSide, this.move, event)
+    this.repeat = new Repeat(hand, boolSide, this.move, event)
+    this.repeatX = new RepeatX(hand, boolSide, this.move, event)
+    this.variable = new Variable(hand, boolSide, this.move, event)
+    this.discard = new Discard(hand, boolSide, this.move, event)
+    this.stackToPlay = undefined
+    this.cardToPlay = undefined
+    this.opponentToAttack = undefined
+    this.moveType = undefined
+  }
+
+  /**
+   * This creates the order of operations for the ai steps and executes them until a step has been chosen.
+   * @param type The type of Ai.
+   */
+  setAi (type) {
+    let moveList = []
+    if (type === 'gambler') {
+      moveList = [this.hack, this.virus, this.powerOutage, this.firewall, this.generator, this.antiVirus, this.overclock,
+        this.batteryBackup, this.variable, this.repeat, this.instruction, this.repeatX, this.group, this.discard]
+    } else if (type === 'hacker') {
+      moveList = [this.hack, this.virus, this.powerOutage, this.firewall, this.generator, this.antiVirus, this.overclock, this.batteryBackup, this.variable, this.repeat,
+        this.instruction, this.repeatX, this.group, this.discard]
+    } else if (type === 'sprinter') {
+      moveList = [this.variable, this.overclock, this.repeatX, this.repeat, this.virus, this.firewall, this.generator, this.antiVirus, this.powerOutage, this.batteryBackup,
+        this.instruction, this.hack, this.group, this.instruction, this.discard]
+    } else if (type === 'protector') {
+      moveList = [this.group, this.firewall, this.generator, this.antiVirus, this.overclock, this.virus, this.powerOutage, this.batteryBackup, this.variable,
+        this.repeat, this.instruction, this.repeatX, this.hack, this.instruction, this.discard]
+    }
+    for (let turn in moveList) {
```

Comment:
As `discard` is always the last action, you could remove it from the list and have it as the action if the for loop falls through (i.e. no other action executed)